lib/bolt/transport/remote.rb

```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+
+require 'bolt/task/remote'
+require 'bolt/transport/base'
+
+module Bolt
+  module Transport
+    class Remote < Base
+      def self.options
+        # TODO: We should accept arbitrary options here
+        %w[host port user password connnect-timeout device-type run-on]
+      end
+
+      def self.validate(options)
+        # This will fail when validating global config
+        # unless options['device-type']
+        #  raise Bolt::ValidationError, 'Must specify device-type for devices'
+        # end
+      end
+
+      # TODO: this should have access to inventory so target doesn't have to
+      def initialize(executor)
+        super()
+
+        @executor = executor
+      end
+
+      def get_proxy(target)
+        # TODO: This needs to have access to the inventory
+        inventory = target.inventory
+        raise "Target was created without inventory? Not get_targets?" unless inventory
+        inventory.get_targets(target.options['run-on'] || 'localhost').first
+      end
+
+      # Cannot batch because arugments differ
+      def run_task(target, task, arguments, options = {})
+        proxy_target = get_proxy(target)
+        transport = @executor.transport(proxy_target.protocol)
+        arguments = arguments.merge('_target' => target.to_h.reject { |_, v| v.nil? })
+
+        remote_task = Bolt::Task::Remote.new(task.to_h)
+
+        # TODO: should we have to handle this logic
```

Comment:
Does this really make sense as a transport? We're having to break a lot of the abstraction to make it work.

Reply:
I had a previous implementation in the executor/inventory more directly but it was a lot more code and made the inventory syntax less natural. I'm open to suggestions if you want to talk about it.